python/lsst/sconsUtils/state.py

```diff
@@ -128,6 +130,11 @@ def _initEnvironment():
                 ourEnv[varname] = os.environ[varname]
                 ourEnv[k] = os.environ[k]
 
+    # add <build root>/ups directory to the configuration search path
+    # this allows the .cfg file for the package being built to be found without
+    # requiring <product name>_DIR to be in the env
+    cfgPath.append(os.path.join(SCons.Script.Dir('#').abspath, 'ups'))
```

Comment:
I can't think of any examples in our own codebase where this happens (which doesn't mean they don't exist), but there are places in the third-party packages where this can happen.